tinyDA/proposal.py

```diff
@@ -40,6 +41,30 @@ def get_q(self, x_link, y_link):
         pass
 
 
+class SharedArchiveProposal(Proposal):
+    def __init__(self):
+        # so that the archive can tell what chain is the sample's origin
+        self.id = None
+        self.archive_reference = None
+
+    def set_id(self, proposal_id):
+        self.id = proposal_id
+
+    def link_archive(self, archive_reference):
+        self.archive_reference = archive_reference
+
+    def read_archive(self):
+        # to get the value and block the thread until its ready
+        archive = ray.get(self.archive_reference.get_archive.remote())
+        #print(len(archive)) #debug
+        return archive
+
+    def update_archive(self, params):
+        # ray.get here to wait for the sample to be added
+        # if there was no ray.get, this sample could be excluded from the next read_archive call in the same chain
+        ray.get(self.archive_reference.update_archive.remote(params, self.id))
```

Comment:
I don't think you need to use `ray.get` here, since the method isn't returning anything.